doc: Bluetooth: mesh: add LCD documentation (#58139)
zephyrproject-rtos/zephyr: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst

.. _bluetooth_mesh_lcd_srv:

Large Composition Data Server
#############################

The Large Composition Data Server model is a foundation model defined by the Bluetooth
mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` option.

The Large Composition Data Server model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and is used to support

> the functionality of exposing pages of composition data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.

Reviewer: ```suggestion
the functionality of exposing pages of Composition Data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
```

> Metadata contains information about the models instantiated on the element. The :c:member:`bt_mesh_models_metadata_entry` struct can be used to define a list of metadata entries.

Reviewer: ```suggestion
Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
```

> Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and id of the metadata, and a pointer to the raw data.

Reviewer: ```suggestion
Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
```

> Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.

Reviewer: ```suggestion
Entries should be created using the :c:macro:`BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
```

> If the list is empty, the helper macro `BT_MESH_MODELS_METADATA_NONE` can be used instead.

Reviewer: ```suggestion
If the list is empty, the helper macro :c:macro:`BT_MESH_MODELS_METADATA_NONE` can be used instead.
```

> The `BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.

Reviewer: I somehow managed to miss this yesterday... Shouldn't these macros be written with :c:macro:`BT_MESH_MODELS_METADATA_ENTRY`, etc?

Reviewer: These comments render strange sometimes. I am using ` in front and end of the name.

> The `BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.

Reviewer: ```suggestion
The :c:macro:`BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.
```

> Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
> Entries should be created using the :c:macro:`BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
> If the list is empty, the helper macro :c:macro:`BT_MESH_MODELS_METADATA_NONE` can be used instead.
> The :c:macro:`BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.

Reviewer 2: ```suggestion
Models metadata
===============

The Large Composition Data Server model allows each model to have a list of model's specific metadata that can be read by the Large Composition Data Client model. The metadata list can be associated with the :c:struct:`bt_mesh_model` through the :c:member:`bt_mesh_model.metadata` field. The metadata list consists of one or more entries defined by the :c:struct:`bt_mesh_models_metadata_entry` structure. Each entry contains the length and ID of the metadata, and a pointer to the raw data. Entries can be created using the :c:macro:`BT_MESH_MODELS_METADATA_ENTRY` macro. The :c:macro:`BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present. If the model has no metadata, the helper macro :c:macro:`BT_MESH_MODELS_METADATA_NONE` can be used instead.
```

And move it below line 20.

The Large Composition Data Server does not have an API of its own and relies on a :ref:`bluetooth_mesh_lcd_cli` to control it.
The model only accepts messages encrypted with the node's device key.

> If present, the Large Composition Data Server model must be instantiated on the primary element.

Reviewer 3: ```suggestion
If present, the Large Composition Data Server model must only be instantiated on the primary element.
```
